### ISSUE TITLE
setlocale() errors with wxPython 3.0.2

### ISSUE THREAD

**Issue by wwwald** [WikidPad/WikidPad#8]
I'm noticing an issue when toggling between Edit and Preview mode. This is with the latest version of Wikidpad (2.3beta13_01), running on Python 2.7.11 with wxPython 3.0.2.0.

A first switch to Preview mode still works, but a second switch to Preview mode doesn't anymore. Everytime I try to switch, I get this error message in the log:
```
Traceback (most recent call last):
  File "lib\pwiki\PersonalWikiFrame.py", line 1648, in <lambda>
    lambda evt: self.setDocPagePresenterSubControl(None),  "tb_switch ed prev",
  File "lib\pwiki\PersonalWikiFrame.py", line 5161, in setDocPagePresenterSubControl
    presenter)
  File "lib\pwiki\MainAreaPanel.py", line 566, in switchDocPagePresenterTabEditorPreview
    presenter.switchSubControl("preview", gainFocus=True)
  File "lib\pwiki\DocPagePresenter.py", line 421, in switchSubControl
    subControl.setLayerVisible(True)
  File "lib\pwiki\WikiHtmlView.py", line 186, in setLayerVisible
    self.refresh()
  File "lib\pwiki\WikiHtmlView.py", line 290, in refresh
    for s in self._DEFAULT_FONT_SIZES])
  File "C:\Users\standaea\Envs\wikidpad\lib\site-packages\wx-3.0-msw\wx\html.py", line 1031, in SetFonts
    return _html.HtmlWindow_SetFonts(*args, **kwargs)
PyAssertionError: C++ assertion "strcmp(setlocale(LC_ALL, NULL), "C") == 0" failed at ..\..\src\common\intl.cpp(1449) in wxLocale::GetInfo(): You probably called setlocale() directly instead of using wxLocale and now there is a mismatch between C/C++ and Windows locale.
Things are going to break, please only change locale by creating wxLocale objects to avoid this!
```
Seems to be a problem with locale setting. I haven't been able to dive into the code yet, but I guess reporting this couldn't hurt

**Comment by gchriz** [comment]
I can't replicate that here.
(Wikidpad 2.3beta13_01+, Python 2.7.10, wxGTK3/wxPython 3.0.2, Fedora 23).

Have you tried with a completely fresh Wiki too?

WikidPad isn't really ready for wxPython 3.x. There are several problems to be solved.

But isn't this one more likely an issue in wxPython rather than in WikidPad?
See for example the second message here: http://wxpython-users.1045709.n5.nabble.com/Fail-on-trying-to-load-png-PyAssertionError-td5723906.html

**Comment by gchriz** [comment]
Oops, sorry, wrong button...
Reopened.

**Comment by wwwald** [comment]
On the right track: this does NOT occur with a fresh wiki. I'll try rebuilding my wiki database for the wiki that's showing the problem, and see what happens.

Just curious... can you give an example of such a WikidPad issue specific to wxPython 3? I've been running it for quite a while without obvious or blocking problems... 
The problem above is actually the first I've noticed that seems to be wxPython-related.

**Comment by wwwald** [comment]
Fixed in commit 625e0e6, closing.
